Files/Helpers/StorageItemHelpers.cs

```diff
@@ -20,35 +22,83 @@ public static async Task<IStorageItem> ToStorageItem(this IStorageItemWithPath i
             FilesystemResult<StorageFile> file = null;
             FilesystemResult<StorageFolder> folder = null;
 
-            if (associatedInstance == null)
+            if (path.EndsWith(".lnk") || path.EndsWith(".url"))
             {
-                file = await FilesystemTasks.Wrap(() => StorageFileExtensions.DangerousGetFileFromPathAsync(path));
+                // TODO: In the future, when IStorageItemWithPath will inherit from IStorageItem,
+                //      we could implement this code here for getting .lnk files
+                //      for now, we can't
+                
+                return default(TOut);
 
-                if (!file)
+                if (false) // Prevent unnecessary exceptions
```

Comment:
This condition will never be true, is that intended?

Reply:
Yes, it is intended to let the person debugging the code know that it got bad arguments and would throw an exception. It will be removed in the future when `IStorageItemWithPath` will implement `IStorageItem`